app/controllers/ems_common.rb

```diff
@@ -678,6 +720,70 @@ def refreshemss
     end
   end
 
+  def call_ems_pause(emss)
+    process_emss(emss, "pause_ems") unless emss.empty?
+    return if @flash_array.present?
+
+    add_flash(n_("Disable initiated for %{count} %{model} from the %{product} Database",
+                 "Disable initiated for %{count} %{models} from the %{product} Database", emss.length) %
+                {:count   => emss.length,
+                 :product => I18n.t('product.name'),
+                 :model   => ui_lookup(:table => table_name),
+                 :models  => ui_lookup(:tables => table_name)})
+  end
+
+  def call_ems_resume(emss)
+    process_emss(emss, "resume_ems") unless emss.empty?
+    return if @flash_array.present?
+
+    add_flash(n_("Resume initiated for %{count} %{model} from the %{product} Database",
+                 "Resume initiated for %{count} %{models} from the %{product} Database", emss.length) %
+                {:count   => emss.length,
+                 :product => I18n.t('product.name'),
+                 :model   => ui_lookup(:table => table_name),
+                 :models  => ui_lookup(:tables => table_name)})
+  end
+
+  def pauseemss
+    assert_privileges(params[:pressed])
+    if @lastaction == "show_list"
+      emss = find_checked_items
+      if emss.empty?
+        add_flash(_("No %{model} were selected for pause") % {:model => ui_lookup(:table => table_name)}, :error)
+      end
+      call_ems_pause(emss)
+      show_list
+      @refresh_partial = "layouts/gtl"
+    else
+      if params[:id].nil? || model.find_by_id(params[:id]).nil?
+        add_flash(_("%{record} no longer exists") % {:record => ui_lookup(:table => table_name)}, :error)
+      else
+        call_ems_pause([params[:id]])
+      end
+      params[:display] = @display
+    end
+  end
+
+  def resumeemss
```

Comment:
Oh yes please :)

`refreshems` and `pauseems` seem to differ only in the flash message and calling `call_ems_*`.

`call_ems_*` differ only in a flash message and an argument to `process_emss`.

And even those ` elsif task == "resume_ems"` and ` elsif task == "pause_ems"` differ only in the flash/audit message and calling `disable!` vs `enable!`. 

So, definitely for DRYing this up a little ;).

Also, I notice that both `call_ems_*` and those `elsif task == ..` bits use the same exact flash messages, so maybe those should be shared somehow as well.... oh and there's a difference where `call_ems_pause` says `Disable initiated` while task `pause_ems` says `Pause initiated` .. which is probably a bug, right? :)